spidermon/contrib/scrapy/monitors/monitors.py

```diff
@@ -563,7 +571,13 @@ def _get_jobs(self, states, number_of_jobs):
             filters=dict(has_tag=tags) if tags else None,
         )
         while _jobs:
-            jobs.extend(_jobs)
+            if close_reason:
+                for job in _jobs:
+                    if job.get("close_reason") in close_reason:
+                        jobs.append(job)
+            else:
+                jobs.extend(_jobs)
```

Comment:
**suggestion (non-blocking):** It should be alright to use `jobs.append()` for both cases to simplify a bit the code flow.

```python
while jobs:
    for job in _jobs:
        if close_reason and job.get("close_reason") not in close_reason:
            continue
        jobs.append(job)
    ...
```